tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -9,72 +9,89 @@
 namespace Quantum.Kata.RandomNumberGeneration {
     
     open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
     open Microsoft.Quantum.Diagnostics;
     open Microsoft.Quantum.Convert;
     open Microsoft.Quantum.Math;
     open Microsoft.Quantum.Arrays;
     open Quantum.Kata.Utils;
     open Microsoft.Quantum.Random;
 
+    // ------------------------------------------------------
+    /// # Summary
+    /// Helper operation to rerun test operation several times
+    /// (a single run can fail with non-negligible probability even for a correct solution).
+    /// # Input
+    /// ## testOp
+    /// Helper test operation which verifies the user answer and reference implementation
+    operation RetryTestOperation (testOp : (Unit => Bool)) : Unit {
+        let numRetries = 3;
+        mutable sufficientlyRandom = false;
+        mutable attemptNum = 1;
+        repeat {
+            set sufficientlyRandom = testOp();
+            set attemptNum += 1;
+        } until (sufficientlyRandom or attemptNum > numRetries);
+
+        if not sufficientlyRandom {
+            fail $"Failed to generate sufficiently random integer";
+        }
+    }
+
     // ------------------------------------------------------
     // Exercise 1.
     @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T1_RandomBit () : Unit {
         Message("Testing one random bit generation...");
-        RetryCheckUniformDistribution(RandomBit_Wrapper, 0, 1, 1000);
+        let userOp = RandomBit;
```

Comment:
It took me a bit of time to figure out which operation corresponds to what part of the testing process; minor suggestion, but I'd recommend to call these three operations "solution", "testingHarness" and "delayedTesting" (in other tests as well).

Reply:
Done. Just made a small minor change. I was not aware of the fact that Q# assumes that any set of parameters is a tuple, and converts it to/from a tuple automatically. So, this allowed me to merge "testingHarness" and "delayedTesting" into "delayedTesting".